### PR TITLE
column.key: string

### PR DESCRIPTION
**Pull request by nstepien**

### CODE REVIEW

File: src/formatters/ValueFormatter.tsx [hunk 1]
```diff
@@ -2,5 +2,5 @@ import React from 'react';
 import { FormatterProps } from '../common/types';
 
 export function ValueFormatter<R>(props: FormatterProps<R>) {
-  return <>{props.row[props.column.key]}</>;
+  return <>{props.row[props.column.key as keyof R]}</>;
```
Comment (nstepien) [inline src/formatters/ValueFormatter.tsx]: IMO we should remove the default defaultFormatter, and delete `ValueFormatter`. Columns should just do
```tsx
formatter(props) { return <>{props.row.specificValue}</>; }
```

Comment (amanmahajan7) [reply src/formatters/ValueFormatter.tsx]: we can consider it

File: src/utils/columnUtils.ts [hunk 2]
```diff
@@ -15,7 +15,7 @@ export function getColumnMetrics<R>(metrics: Metrics<R>): ColumnMetrics<R> {
   let allocatedWidths = 0;
   let unassignedColumnsCount = 0;
   let lastFrozenColumnIndex = -1;
-  const columns: Array<Column<R> & { width: number | void }> = [];
+  const columns: Array<Column<R> & { width: number | undefined }> = [];
```
Comment (amanmahajan7) [inline src/utils/columnUtils.ts]: `void` was failing for some reason

Comment (qili26) [reply src/utils/columnUtils.ts]: I think `void` means no data type, `undefined` is a valid `nothing exists` type. They are still slightly different.

File: src/common/types.ts [hunk 3]
```diff
@@ -37,13 +37,12 @@ export interface Column<TRow> {
...
 }
 
-export type CalculatedColumn<TRow> =
-  Column<TRow> & {
-    idx: number;
-    width: number;
-    left: number;
-    formatter: React.ComponentType<FormatterProps<TRow>>;
-  };
+export interface CalculatedColumn<TRow> extends Column<TRow> {
```
Comment (amanmahajan7) [inline src/common/types.ts]: Do we need to omit `Column<Row>.formatter`?

Comment (nstepien) [reply src/common/types.ts]: Doesn't look like it, it's overriding both width and formatter accordingly.

**Pull request merged**